include/envoy/stats/stats.h

```diff
@@ -281,5 +281,29 @@ class StoreRoot : public Store {
 
 typedef std::unique_ptr<StoreRoot> StoreRootPtr;
 
+struct RawStatData;
+
+/**
+ * Abstract interface for allocating a RawStatData.
+ */
+class RawStatDataAllocator {
+public:
+  virtual ~RawStatDataAllocator() {}
+
+  /**
+   * @return RawStatData* a raw stat data block for a given stat name or nullptr if there is no more
+   *         memory available for stats. The allocator may return a reference counted data location
```

Comment:
Technically, I think s/may return/should return/. We have an outstanding issue related to this at https://github.com/envoyproxy/envoy/issues/2453